plugin/db/mysql/mysql.go

```diff
@@ -34,15 +35,17 @@ type Driver struct {
 	dbType        db.Type
 	resourceDir   string
 	binlogDir     string
+	binlogMetaDir string
 	db            *sql.DB
 
 	replayBinlogCounter *common.CountingReader
 }
 
 func newDriver(dc db.DriverConfig) db.Driver {
 	return &Driver{
-		resourceDir: dc.ResourceDir,
-		binlogDir:   dc.BinlogDir,
+		resourceDir:   dc.ResourceDir,
+		binlogDir:     dc.BinlogDir,
+		binlogMetaDir: filepath.Join(dc.BinlogDir, "metadata"),
```

Comment:
Does it have to be a separate directory?

Reply:
With a separate director (same prefix for path), we can easily fetch all the metadata for an instance with the list API of cloud storage.
The downside is that we need to deal with the "metadata" directory when traversing the binlog files.

Reply:
"fetch all the metadata for an instance with the list API of cloud storage." is it listing all objects in a directory and download them?

Reply:
> "fetch all the metadata for an instance with the list API of cloud storage." is it listing all objects in a directory and downloading them?

Exactly.

Reply:
Then putting binary logs and their metadata files in the same directory would be a better choice. We only need to write a file name matching function, right?

Reply:
SGTM. I'll turn to this way.